src/sparseml/transformers/sparsification/obcq/sparse_gpt_modifier.py

```diff
@@ -0,0 +1,130 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from typing import Optional
+
+import torch
+
+from sparseml.pytorch.sparsification.modifier import BaseModifier, PyTorchModifierYAML
+from sparseml.transformers.sparsification.obcq.layer_compressor import LayerCompressor
+
+
+__all__ = ["SparseGPTModifier"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@PyTorchModifierYAML()
+class SparseGPTModifier(BaseModifier):
+    def __init__(
+        self,
+        sparsity: float = 0.5,
+        block_size: int = 128,
+        quantize: bool = True,
+        dampening_frac: Optional[float] = 0.01,
+        sequential_update: Optional[bool] = True,
+    ):
+        super().__init__()
+
+        self._compressible_layers = None
+        self._model = None
+        self._bottom_compressor = None
+        self._head_compressor = None
+        self._compressible_layers = None
+
+        self._sparsity = sparsity
+        self._block_size = block_size
+        self._quantize = quantize
+        self._dampening_frac = dampening_frac
+        self._sequential_update = sequential_update
+
+        self._device = "cuda:0"
+
+    def compressible_layers(self):
+        raise NotImplementedError
+
+    def bottom_compressor(self):
+        raise NotImplementedError
+
+    def head_compressor(self):
+        raise NotImplementedError
+
+    def one_shot(self, model, dataloader, initializer_kwargs, finalize_kwargs):
```

Comment:
would be great to move all this functionality into just initialize and finalize, so at the generic modifier/manager level, one shot can be applied by just calling these two